src/models/recipe/recipe.schema.ts

```diff
@@ -0,0 +1,35 @@
+import { Schema, model } from "mongoose";
+import mongoose from "mongoose";
+import { IRecipeDocument, IRecipeModel } from "./recipe.types";
+
+//Recipe Schema
+const RecipeSchema: Schema = new Schema(
+  {
+    name: {
+      type: String,
+      required: true,
+    },
+    desription: {
+      type: String,
+      required: true,
+    },
+    likes: {
+      key: String,
+      ref: "user",
+      value: Date,
+    },
+    postedBy: {
+      type: String,
+      ref: "user",
+    },
+    images: {
+      type: [{ url: String }],
+    },
+  },
+  {
+    timestamps: true,
+  }
```

Comment:
With experience in doing this, and knowing that it works, you can update the timestamps to this
```suggestion
{
    timestamps: true,
    strict: false,
    typePojoToMixed: false,
  } as SchemaOptionsWithPojoToMixed
```
And then add this before you define the Schema (which will stop the TS complaining)
```
interface SchemaOptionsWithPojoToMixed extends SchemaOptions {
  typePojoToMixed: boolean;
}
```